### PR TITLE
Add PyAthena Dialect

### PR DESCRIPTION
**Pull request by jasonamyers**
### Description
- Adds PyAthena to list of external dialects

### Checklist

This pull request is:

- [X] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

### CODE REVIEW

**Comment by jvanasco** [comment]
LGTM, just a docs change. 

@CaselIT does this even need to go into Gerrit?

**Comment by CaselIT** [comment]
Thanks

> LGTM, just a docs change.
> 
> @CaselIT does this even need to go into Gerrit?

No, I'll usually merge it locally. I'll try to this later.

File: doc/build/dialects/index.rst [hunk 1]
```diff
@@ -67,6 +67,8 @@ Currently maintained external dialect projects for SQLAlchemy include:
 +------------------------------------------------+---------------------------------------+
 | Amazon Redshift (via psycopg2)                 | sqlalchemy-redshift_                  |
 +------------------------------------------------+---------------------------------------+
+| Amazon Athena                                  | pyathena_                             |
```
Comment (CaselIT) [inline doc/build/dialects/index.rst:70]: since this table is more or less in alphabetical order, can you move this above redshift?

**Review by CaselIT — CHANGES_REQUESTED** [review]

File: doc/build/dialects/index.rst (hunk 1 above)
Comment (jasonamyers) [reply doc/build/dialects/index.rst:68]: Done sir

**Review by CaselIT — APPROVED** [review]

**Pull request merged**